2023-07-14 Albert Larsan (#221)
rustacean-station/rustacean-station.org: _episodes/interviews/2023-07-14-albert-larsan.md

---
title: "Bootstrapping Rust with Albert Larsan"

> date: 2023-07-14T13:01:00Z

Reviewer: ```suggestion
date: 2023-07-14T14:08:00Z
```

file: https://audio.rustacean-station.org/file/rustacean-station/2023-07-14-albert-larsan.mp3
duration: "33:04"
length: "31745254"
#reddit: (leave blank on initial publish, amend with link and uncomment this line after Reddit thread has been posted)
---

> Allen Wyma talks with [Albert Larsan](https://twitter.com/cernjai) about his work on bootstrapping the Rust compiler.

Reviewer 2: ```suggestion
Allen Wyma talks with [Albert Larsan](https://github.com/albertlarsan68) about his work on bootstrapping the Rust compiler.
```

## Contributing to Rustacean Station

Rustacean Station is a community project; get in touch with us if you'd like to suggest an idea for an episode or offer your services as a host or audio editor!

- Twitter: [@rustaceanfm](https://twitter.com/rustaceanfm)
- Discord: [Rustacean Station](https://discord.gg/cHc3Gyc)
- Github: [@rustacean-station](https://github.com/rustacean-station/)
- Email: [hello@rustacean-station.org](mailto:hello@rustacean-station.org)

## Timestamps

- [@0:00] - Introduction to Bootstrap
- [@1:57] - Building the standard library and compiler.

> - [@5:34] - The biggest challenges of bootstrapping RustC

Reviewer 2: ```suggestion
- [@5:34] - The biggest challenges of bootstrapping Rustc
```
I think this is the correct case, I've never seen it written as RustC.

> - [@5:34] - The biggest challenges of bootstrapping Rustc

Reviewer: ```suggestion
- [@5:34] - The biggest challenges when bootstrapping `rustc` (the Rust compiler)
```

- [@11:26] - Why use Python to start the bootstrapping process?

> - [@13:08] - Running tests as part of the Rust API, ensuring that each component is thoroughly tested.

Reviewer: ```suggestion
- [@13:08] - Running tests as part of the Rust API to ensure that each component is well-tested.
```

> - [@15:07] - Running tests on Linux, MacOS & Windows

Reviewer: ```suggestion
- [@15:07] - Running tests on Linux, MacOS, and Windows
```

> - [@18:54] - Features that the bootstrapping team for RustC has been working on.

Reviewer: ```suggestion
- [@18:54] - Features that the `rustc` bootstrapping team has been working on.
```

> - [@18:54] - Features that the `rustc` bootstrapping team has been working on.

Reviewer 2: ```suggestion
- [@18:54] - Features that the Rust bootstrap team has been working on.
```

The name of the team is bootstrap, not bootstrapping

Reviewer: Thanks!

- [@20:50] - Plans to run Clippy and fix issues.
- [@23:41] - Stage redesign, verification, reproducibility, and different optimizations implemented in the compiler.

> - [@27:28] - Albert's advice on the process of bootstrapping Rustc.

Reviewer: ```suggestion
- [@27:28] - Albert's advice on the process of bootstrapping Rust.
```

> - [@31:05] - Importance of documentation for Rustc

Reviewer: ```suggestion
- [@31:05] - The importance of documentation
```
